module_utils/network_lsr/argument_validator.py

```diff
@@ -1802,3 +1802,23 @@ def validate_connection_one(self, mode, connections, idx):
                     idx,
                     "ip.ipv6_disabled is not supported by initscripts.",
                 )
+        # when IPv4 or IPv6 is disabled, settings for ip.dns_options and
+        # ip.dns_search are not allowed
+        if connection["ip"]["dns_options"] or connection["ip"]["dns_search"]:
+            addrs4 = list(
+                [
+                    address
```

Comment:
minor comment: here you use a list comprehension with `[]`. That already generates a list. The code `list([ ... ])` generates a second list (throwing away the first). Maybe python can optimize that, IDK. But it also is verbose to read.

Also you could write `any_addrs4 = any( [list comprehension] )` as you don't actually need the entire list `addrs4`. When you do that, you could use a generator comprehension instead of a list comprehension. The difference is that `any()` only needs to iterate until it find one true value. It does not need the entire list, and neither does it need to generate the enitre list in the first place.

Meaning, I'd do:

```python
has_addrs4 = any( a["family"] == AF_INET for a in connection["ip"]["address"] )
```

Reply:
Then the `list()` here is also redundant,
https://github.com/linux-system-roles/network/blob/main/library/network_connections.py#L972

Reply:
it is!

Reply:
I changed into
```
            has_addrs4 = any(
                address["family"] == socket.AF_INET
                for address in connection["ip"]["address"]
            )
```